EIPS/eip-4519.md

```diff
@@ -0,0 +1,239 @@
+ 
+**User Management Mechanism**
+ 
+SmartNFTs allow implementing a new and useful user management mechanism. In the last few years, many projects concerning assets sharing (for example, vehicles) have been created and developed. We incorporate the blockchain account of the user of the token as another attribute in order to distinguish between users, who employ the asset for a specific application, and owners, who assign the token to users. Both users and owners can be traced by the blockchain.
+ 
+**Secure Key Exchange Mechanism**
+ 
+The engagements of the asset with an owner and a user are carried out after a mutual authentication protocol (for example, based on elliptic curve Diffie-Hellman key exchange protocol). This protocol can be employed for a key agreement between the asset and its owner, in the one side, and the asset and its user, in the other side. 
+ 
+When the SmartNFT is created or when the ownership is transferred, the operating mode of the asset defined by the token state is "Waiting for owner". Assuming that the asset is an electronic physical asset, it saves in its memory the owner address. The owner generates a pair of keys using the elliptic curve secp256k1 and the primitive element P used on this curve: a secret key (SK<sub>O_D</sub>) and a Public Key (PK<sub>O_D</sub>), so that PK<sub>O_D</sub> = SK<sub>O_D</sub>*P. To generate the shared key between owner and asset, (K<sub>O</sub>), the public key of the asset, (PK<sub>DEV</sub>), is employed as follows:
+ 
+K<sub>O</sub>=PK<sub>DEV</sub>*SK<sub>O_D</sub>
+
+Using the function startOwnerEngagement, the owner saves PK<sub>O_D</sub> and the hash of K<sub>O</sub> in the SmartNFT. The owner sends PK<sub>O_D</sub> signed to the asset. The asset checks the signature to verify the identity of the owner and calculates:
+ 
+K<sub>O_D</sub> = SK<sub>DEV</sub>*PK<sub>O_D</sub>
+
+If everything is correctly done, K<sub>O</sub> and K<sub>O_D</sub> are the same since:
+ 
+K<sub>O</sub>=PK<sub>DEV</sub>*SK<sub>O_D</sub>=(SK<sub>DEV</sub>*P)*SK<sub>O_D</sub>= SK<sub>DEV</sub>*(SK<sub>O_D</sub>*P)=SK<sub>DEV</sub>*PK<sub>O_D</sub>
+
+Using the function ownerEngagement, the asset sends the K<sub>O_D</sub> obtained and if it is the same as K<sub>O</sub>, then the state of the token changes to "Engaged with owner" and the event OwnerEngaged is sent. Once the asset receives the event, it changes its operation mode to "Engaged with owner". This process is shown in Figure 4. From this moment, the asset can be managed by the owner. 
```

Comment:
I _think_ this is probably specification? It doesn't seem to describe "why" something is, but "what" something is supposed to do.